aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -805,9 +855,14 @@ internal class RealAWSCognitoAuthPlugin(
 
     override fun rememberDevice(onSuccess: Action, onError: Consumer<AuthException>) {
         authStateMachine.getCurrentState { authState ->
-            when (authState.authNState) {
+            when (val state = authState.authNState) {
                 is AuthenticationState.SignedIn -> {
-                    updateDevice(null, DeviceRememberedStatusType.Remembered, onSuccess, onError)
+                    updateDevice(
+                        (state.deviceMetadata as DeviceMetadata.Metadata).deviceKey,
```

Comment:
unsafe cast, this might crash if cast fails
`(state.deviceMetadata as? DeviceMetadata.Metadata)?.deviceKey`

Reply:
This cast can not fail as they are the same type.

Reply:
But I have added your suggestion